com.unity.render-pipelines.universal/Editor/ShaderGUI/BaseShaderGUI.cs

```diff
@@ -325,19 +404,53 @@ protected static void DrawTileOffset(MaterialEditor materialEditor, MaterialProp
         ////////////////////////////////////
         #region MaterialDataFunctions
 
-        public static void SetMaterialKeywords(Material material, Action<Material> shadingModelFunc = null, Action<Material> shaderFunc = null)
+        // this function is shared with ShaderGraph Lit/Unlit GUIs and also the hand-written GUIs
+        internal static void UpdateMaterialSurfaceOptions(Material material)
         {
             // Setup blending - consistent across all Universal RP shaders
             SetupMaterialBlendMode(material);
 
+            bool isShaderGraph = material.IsShaderGraph();
+
+            // Cast Shadows
+            bool castShadows = true;
+            if (material.HasProperty(Property.CastShadows))
+            {
+                castShadows = (material.GetFloat(Property.CastShadows) != 0.0f);
+            }
+            else
+            {
+                if (isShaderGraph)
+                {
+                    // Lit.shadergraph or Unlit.shadergraph, but no material control defined
+                    // enable the pass in the material, so shader can decide...
+                    castShadows = true;
+                }
+                else
+                {
+                    // Lit.shader or Unlit.shader -- set based on transparency
+                    castShadows = Rendering.Universal.ShaderGUI.LitGUI.IsOpaque(material);
+                }
```

Comment:
Thanks for handling this. I'll have a discussion with @Verasl I think URP team should align with ShaderGraph (we can do in follow up PR)